src/components/Crud.tsx

```diff
@@ -1,237 +1,198 @@
-      surname: surname,
+      name: userInputs.name,
+      surname: userInputs.surname,
       id: id,
     };
-    setNameList([...nameList, newName]);
-    setCurrentNameList([...nameList, newName]);
-    // is there a way to re-render only in one time? is it necessary
-    setfilterValue("");
+    setUserList([...userList, newName]);
     setSelectedId(id);
+    resetState("filterValue");
     setMessage("The name is created");
   }
 
   function handleUpdate(selectedId: string | undefined) {
     if (!selectedId) {
-      setMessage("No name is selected");
+      setMessage("Please select a name");
       return;
     }
-    if (isNameInvalid(frontName) || isNameInvalid(surname)) {
-      setMessage("Please give the richt form of name");
+    if (!userInputs.name && !userInputs.surname) {
+      setMessage("Please give a name");
       return;
     }
-    if (selectedId && frontName && surname) {
-      const updatedName = {
-        frontName: frontName,
-        surname: surname,
-        id: selectedId,
-      };
-      const updatedNameList = nameList.map((name) => {
-        return name.id === selectedId ? updatedName : name;
-      });
-      setNameList(updatedNameList);
-      setCurrentNameList(updatedNameList);
-      setfilterValue("");
+    if (selectedId && (userInputs.name || userInputs.surname)) {
+      const oldUserToUpdate = userList.find((user) => user.id === selectedId)!;
+      oldUserToUpdate.name = userInputs.name;
+      oldUserToUpdate.surname = userInputs.surname;
+      resetState("filterValue");
       setMessage("The name is updated");
     }
   }
 
   function handleDelete(selectedId: string | undefined) {
     if (!selectedId) {
-      setMessage("No name selected");
+      setMessage("Please select a name");
       return;
     }
-    const updatedNameList = nameList.filter((name) => name.id !== selectedId);
-    setNameList(updatedNameList);
-    setCurrentNameList(updatedNameList);
-    setSelectedId(undefined);
-    setfilterValue("");
-    setMessage("The name is deleted");
-  }
-
-  function handleChangeNameInput(event: React.ChangeEvent<HTMLInputElement>) {
-    setFrontName(formatAsName(event.target.value));
+    const indexToDelete = userList.findIndex((user) => user.id === selectedId);
+    if (indexToDelete !== -1) userList.splice(indexToDelete, 1);
+    resetState("selectedId");
+    resetState("userInputs");
+    resetState("filterValue");
+    setMessage("The user is deleted");
   }
 
-  function handleChangeSurnameInput(
-    event: React.ChangeEvent<HTMLInputElement>
-  ) {
-    setSurname(formatAsName(event.target.value));
+  function handleFilterChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setFilterValue(e.target.value);
+    const isSelectedIdInFilteredUserList = filterUserList(
+      e.target.value,
+      userList
+    ).some((user) => user.id === selectedId);
+    if (!isSelectedIdInFilteredUserList) {
+      resetState("selectedId");
+      resetState("userInputs");
+    }
```

Comment:
Hier, um zu entscheiden, ob wir das State zurücksetzen wollen, müssen wir:

1. Für jede Element in `userList`, prüfen, ob es Teil des aktuellen Filters ist
2. Für jede Element in der gefilterte Liste, prüfen, ob es das gleiche ID hat wie das `selectedId`.

Gibt es eine schnellere Lösung hier?  Also ein Weg, wo wir in einem Schritt prüfen können, ob der ausgewählte User teil des Filters ist?